core/src/main/scala/org/apache/spark/rdd/ZippedWithIndexRDD.scala

```diff
@@ -64,8 +64,14 @@ class ZippedWithIndexRDD[T: ClassTag](prev: RDD[T]) extends RDD[(T, Long)](prev)
 
   override def compute(splitIn: Partition, context: TaskContext): Iterator[(T, Long)] = {
     val split = splitIn.asInstanceOf[ZippedWithIndexRDDPartition]
-    firstParent[T].iterator(split.prev, context).zipWithIndex.map { x =>
-      (x._1, split.startIndex + x._2)
+    val parentIter = firstParent[T].iterator(split.prev, context)
+    new Iterator[(T, Long)] {
+      var idxAcc: Long = -1L
```

Comment:
also why don't we initialize this to split.startIndex?

I'd also rename this just "index"